org.scala-ide.sdt.core/src/org/scalaide/core/internal/compiler/ScalaPresentationCompiler.scala

```diff
@@ -458,9 +450,26 @@ class ScalaPresentationCompiler(private[compiler] val name: String, _settings: S
   }
 
   private [core] def defaultHyperlinkLabel(sym: Symbol): String = s"${sym.kindString} ${sym.fullName}"
+
+  import ScalaPresentationCompiler.RefTimeStamp
+  import ScalaPresentationCompiler.Ids
+
+  /*
+   * Meant to ease debugging. These fields allow to easily distinguish different presentation compilers,
+   * for example when examining heap dumps.
+   */
+  val timeStamp = (System.currentTimeMillis() - RefTimeStamp)
+  val id = Ids.getAndIncrement
+
+  override def toString = {
+    s"ScalaPresentationCompiler[$id@$timeStamp]"
```

Comment:
So that I can also see how long a presentation compiler has been around. This is especially useful when tracking memory leaks.